### PR TITLE
Release/alpha 0.0.1

### PR DESCRIPTION
**Pull request by MariyaSin**

### CODE REVIEW

File: src/MessageService.Broker/Consumers/CreateWorkspaceConsumer.cs [hunk 1]
```diff
@@ -1,74 +1,37 @@
...
-            _workspaceUserRepository = workspaceUserRepository;
-            _channelRepository = channelRepository;
-            _channelUserRepository = channelUserRepository;
-            _workspaceMapper = workspaceMapper;
-            _workspaceUserMapper = workspaceUserMapper;
-            _channelMapper = channelMapper;
-            _channelUserMapper = channelUserMapper;
-        }
+    public CreateWorkspaceConsumer(
+        IWorkspaceRepository workspaceRepository,
```
Comment (BelehovEgor) [inline src/MessageService.Broker/Consumers/CreateWorkspaceConsumer.cs:23]: space

File: src/MessageService.Business/Commands/EmailTemplate/EditEmailTemplateCommand.cs [hunk 2]
```diff
@@ -1,86 +1,84 @@
...
+  public class EditEmailTemplateCommand : IEditEmailTemplateCommand
+  {
+    private readonly IAccessValidator _accessValidator;
+    private readonly IEmailTemplateRepository _repository;
+    private readonly IEditEmailTemplateValidator _validator;
+    private readonly IPatchDbEmailTemplateMapper _mapper;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public EditEmailTemplateCommand(
+        IAccessValidator accessValidator,
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/EmailTemplate/EditEmailTemplateCommand.cs:28]: space

File: src/MessageService.Business/Commands/EmailTemplateText/EditEmailTemplateTextCommand.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  public class EditEmailTemplateTextCommand : IEditEmailTemplateTextCommand
+  {
+    private readonly IAccessValidator _accessValidator;
+    private readonly IEmailTemplateTextRepository _repository;
+    private readonly IEditEmailTemplateTextValidator _validator;
+    private readonly IPatchDbEmailTemplateTextMapper _mapper;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public EditEmailTemplateTextCommand(
+        IAccessValidator accessValidator,
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/EmailTemplateText/EditEmailTemplateTextCommand.cs:28]: space

File: src/MessageService.Business/Commands/Keyword/AddKeywordCommand.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+using LT.DigitalOffice.MessageService.Data.Interfaces;
+using LT.DigitalOffice.MessageService.Mappers.Db.Interfaces;
+using LT.DigitalOffice.MessageService.Models.Dto.Requests.ParseEntity;
+using LT.DigitalOffice.MessageService.Validation.Validators.ParseEntity.Interfaces;
+using System;
+
+namespace LT.DigitalOffice.MessageService.Business.Commands.ParseEntity
+{
+  public class AddKeywordCommand : IAddKeywordCommand
+    {
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/Keyword/AddKeywordCommand.cs:15]: space

File: src/MessageService.Business/Commands/Workspace/FindWorkspaceCommand.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+namespace LT.DigitalOffice.MessageService.Business.Commands.Workspace
+{
+  public class FindWorkspaceCommand : IFindWorkspaceCommand
+  {
+    private readonly IWorkspaceRepository _repository;
+    private readonly IShortWorkspaceInfoMapper _mapper;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public FindWorkspaceCommand(
+        IWorkspaceRepository repository,
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/Workspace/FindWorkspaceCommand.cs:21]: space

File: src/MessageService.Business/Commands/Workspace/GetWorkspaceCommand.cs [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+        return null;
+      }
+
+      string errorMessage = "Cannot get images now. Please try again later.";
+      const string logMessage = "Cannot get images with ids: {imageIds}.";
+
+      try
+      {
+        IOperationResult<IGetImagesResponse> response = _rcGetImages.GetResponse<IOperationResult<IGetImagesResponse>>(
+            IGetImagesRequest.CreateObj(imageIds)).Result.Message;
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/Workspace/GetWorkspaceCommand.cs:62]: space

File: src/MessageService.Business/Commands/Workspace/GetWorkspaceCommand.cs [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        return null;
+      }
+
+      string errorMessage = "Cannot get users now. Please try again later.";
+      const string logMessage = "Cannot get users with ids: {usersIds}.";
+
+      try
+      {
+        IOperationResult<IGetUsersDataResponse> response = _rcGetUsers.GetResponse<IOperationResult<IGetUsersDataResponse>>(
+            IGetUsersDataRequest.CreateObj(userIds)).Result.Message;
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/Workspace/GetWorkspaceCommand.cs:94]: space

File: src/MessageService.Mappers/Patch/Interfaces/IPatchDbEmailTemplateMapper.cs [hunk 8]
```diff
@@ -1,12 +1,14 @@
...
-    [AutoInject]
-    public interface IEditDbEmailTemplateMapper
-    {
-        DbEmailTemplate Map(EditEmailTemplateRequest request);
-    }
+  [AutoInject]
+  public interface IPatchDbEmailTemplateMapper
+  {
+    JsonPatchDocument<DbEmailTemplate> Map(
+        JsonPatchDocument<EditEmailTemplateRequest> request);
```
Comment (BelehovEgor) [inline src/MessageService.Mappers/Patch/Interfaces/IPatchDbEmailTemplateMapper.cs:12]: space

File: src/MessageService.Mappers/Patch/PatchDbEmailTemplateMapper.cs [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+      }
+
+      JsonPatchDocument<DbEmailTemplate> dbPatch = new();
+
+      foreach (var item in request.Operations)
+      {
+        if (item.path.EndsWith(nameof(EditEmailTemplateRequest.Type), StringComparison.OrdinalIgnoreCase))
+        {
+          dbPatch.Operations.Add(new Operation<DbEmailTemplate>(
+              item.op, item.path, item.from, (int)Enum.Parse(typeof(EmailTemplateType), item.value.ToString())));
```
Comment (BelehovEgor) [inline src/MessageService.Mappers/Patch/PatchDbEmailTemplateMapper.cs:28]: space

File: src/MessageService.Models.Db/DbChannel.cs [hunk 10]
```diff
@@ -1,55 +1,57 @@
...
-                .HasOne(ch => ch.Workspace)
-                .WithMany(w => w.Channels)
-                .HasForeignKey(ch => ch.WorkspaceId);
-
-            builder
-                .HasMany(ch => ch.Users)
-                .WithOne(chu => chu.Channel);
-        }
+      builder
+          .ToTable(DbChannel.TableName);
```
Comment (BelehovEgor) [inline src/MessageService.Models.Db/DbChannel.cs:38]: space

File: src/MessageService.Models.Db/DbEmailTemplateText.cs [hunk 11]
```diff
@@ -1,50 +1,48 @@
...
-                .Property(ett => ett.Text)
-                .IsRequired();
-
-            builder
-                .HasOne(ett => ett.EmailTemplate)
-                .WithMany(et => et.EmailTemplateTexts)
-                .HasForeignKey(ett => ett.EmailTemplateId);
-        }
+      builder
+          .ToTable(DbEmailTemplateText.TableName);
```
Comment (BelehovEgor) [inline src/MessageService.Models.Db/DbEmailTemplateText.cs:26]: space

File: src/MessageService.Models.Db/DbWorkspace.cs [hunk 12]
```diff
@@ -1,54 +1,56 @@
...
-            builder
-                .HasMany(w => w.Users)
-                .WithOne(wa => wa.Workspace);
-
-            builder
-                .HasMany(w => w.Channels)
-                .WithOne(ch => ch.Workspace);
-        }
+      builder
+          .ToTable(DbWorkspace.TableName);
```
Comment (BelehovEgor) [inline src/MessageService.Models.Db/DbWorkspace.cs:38]: space

**Review by BelehovEgor — APPROVED** [review]

**Review by Cheephot — APPROVED** [review]

**Review by VLiamin — APPROVED** [review]

**Pull request merged**